provisioner/provision.go

```diff
@@ -0,0 +1,50 @@
+package provisioner
+
+import (
+	"fmt"
+
+	"github.com/resin-os/resin-provisioner/defaults"
+	"github.com/resin-os/resin-provisioner/util"
+)
+
+func Provision(appId, apiKey, userId, userName, configPath, domain string, dryRun bool) error {
+	deviceType, err := util.ScanDeviceTypeSlug(defaults.OSRELEASE_PATH)
+	if err != nil {
+		return err
+	}
+
+	config := Config{
+		AppId:            appId,
+		ApiKey:           apiKey,
+		UserId:           userId,
+		UserName:         userName,
+		DeviceType:       deviceType,
+		ApiEndpoint:      fmt.Sprintf("https://api.%s", domain),
+		RegistryEndpoint: fmt.Sprintf("registry.%s", domain),
+		VpnEndpoint:      fmt.Sprintf("vpn.%s", domain),
+	}
```

Comment:
Well what this *used* to do is to actually provision the device  and write the corresponding fields to config.json - the supervisor would then see those and use them.
The advantage of that was that the device would show up in the dashboard immediately while you ran this program - and then it would come up online as the supervisor starts. Plus it made more sense with the name "provisioner" :P